Explore: Use new icons for split pane resizing

gelicia opened the pull request:
The icons introduced to the explore toolbar are exactly like other icons the toolbar already has. This brings in new icons designed by @radiorental and uses those instead to be more distinct.

**Which issue(s) does this PR fix?**:
Fixes #60253 

**Note**:
If you toggle the theme via the command palette, you will see some background colors/borders around the buttons here. I'm not sure if this toolbar is doing something strange with styles or not, but I opened up a ticket here https://github.com/grafana/grafana/issues/60397

gelicia committed:
Add icons, use them for split pane resizing

gelicia commented:
Looks bad on dark mode. Fixing..

Elfo404 commented:
is this the actual icon we want to use? i've never seen hand-drawn style icons in grafana, it really clashes with everything else 🤔 
![Screenshot 2022-12-15 at 14 07 42](https://user-images.githubusercontent.com/1170767/207881844-5b52d313-f3c4-4aed-b14d-6f37de7fe38f.png)

academo commented:
Just to be sure. did you use `yarn generate-icons-bundle-cache-file` inside `packages/grafana-ui` to generate the cache and bundle files?

gelicia commented:
> Just to be sure. did you use `yarn generate-icons-bundle-cache-file` inside `packages/grafana-ui` to generate the cache and bundle files?

Yes - I wouldnt have the patience to renumber those imports by hand :)

gelicia committed:
Change icons

gelicia committed:
Apply scale to child div

gelicia committed:
Use fancier logic for displaying icon

gelicia committed:
Remove color fill on icon

gelicia commented:
https://user-images.githubusercontent.com/1533128/207966026-41ead3fa-8c64-45c6-92e1-67d397bdf2cb.mov

After talking about it, we changed the icons to something slightly different and removed the fill to blend into the theme better.

gelicia committed:
Merge branch 'main' of https://github.com/grafana/grafana into kristina/split-icons

gelicia committed:
Change icons, remove scale

gelicia commented:
@L-M-K-B @Elfo404 Changed the icons again :D

Elfo404 commented on public/app/features/explore/ExploreToolbar.tsx:
```diff
@@ -168,11 +180,9 @@ class UnConnectedExploreToolbar extends PureComponent<Props> {
-                ? 'angle-left'
-                : 'angle-right'
-            }
+            icon={isLargerExploreId ? 'gf-movepane-right' : 'gf-movepane-left'}
+            iconOnly={true}
+            className={styles.rotateIcon}
```
as styles would get recreated and inserted at every render, can't we just switch the icon based on `exploreId` instead of rotating it?

gelicia replied:
The idea here is to keep the arrow in the pane where it's displayed in the icon. So the left pane always has an arrow on the left side, and the icon itself either has the arrow pointing into or away from the smaller pane depending on if we're making the pane larger or not.

Elfo404 replied:
![Screenshot 2022-12-20 at 13 28 19](https://user-images.githubusercontent.com/1170767/208678157-179e3586-a92e-4dfa-9b5c-c172d9e554df.png)

that doesn't seem to be the case though 🤔 

also the arrows in the new icons always points internally, if you rotate one of the icons by 180 deg you simply get the other icon

radiorental replied:
Apologies @gelicia , I didn't pick up on this detail but I think @Elfo404 is correct.  For reference here are the previous icons (below) and the latest (above).
<img width="722" alt="Screenshot 2022-12-20 at 9 23 43 AM" src="https://user-images.githubusercontent.com/3299847/208689718-6ff147ca-a7af-4ee9-86e6-1bd2bdd1354a.png">

radiorental replied:
I modified it if that helps? 
![movepane-rtl(1)](https://user-images.githubusercontent.com/3299847/208690374-ce9478a4-4b89-4e52-ae84-6487ee812c5e.svg)

gelicia replied:
> ![Screenshot 2022-12-20 at 13 28 19](https://user-images.githubusercontent.com/1170767/208678157-179e3586-a92e-4dfa-9b5c-c172d9e554df.png)
> 
> that doesn't seem to be the case though 🤔
> 
> also the arrows in the new icons always points internally, if you rotate one of the icons by 180 deg you simply get the other icon

@radiorental @Elfo404 Ah sorry, I didn't know the assets didn't include both arrows. Fixing.

gelicia committed:
Change icons and logic

Elfo404 reviewed (approved):
Nice stuff! 🚀

radiorental reviewed (approved):
ship it
